.github/workflows/tf-plan-hash.yml

```diff
@@ -0,0 +1,130 @@
+name: HASH Terraform Plan
+
+# TODO: While this workflow is specific to HASH, we should consider
+#  making this a reusable workflow, parameterized on the directory and environment.
+
+on:
+  workflow_dispatch:
+    inputs:
+      env:
+        type: choice
+        options: ["prod"]
+        description: "Environment to plan"
+        required: true
+        default: "prod"
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "infra/terraform/hash/**"
+      - "infra/terraform/modules/**"
```

Comment:
These will prevent making this a required job. I think we want to make it required at some point?

Reply:
I mentioned a workaround in [this slack thread](https://hashintel.slack.com/archives/C02TWBTT3ED/p1684844804416769?thread_ts=1684844296.022789&cid=C02TWBTT3ED): It's possible by either having an empty job running or by using the `if:` conditions: [Handling skipped but required checks](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks). A filtered out path will result in the `Pending` state, while `if: skipped` will result in a `Success` state.

Reply:
Removed the path condition in 9d8175a35d867d98359137e6d3f8636d68a4a1a3 as it wasn't really worth the hassle to do path matching when we can just run the (quick) workflow

Reply:
I think that's fair, thank you!